src/Worker/ReschedulingDecorator.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neighborhoods\KojoWorkerDecoratorComponent\Worker;
+
+use DateTime;
+use Doctrine\DBAL\Connection;
+use LogicException;
+use Neighborhoods\KojoWorkerDecoratorComponent\WorkerInterface;
+use Neighborhoods\ThrowableDiagnosticComponent\ThrowableDiagnostic;
+use OutOfBoundsException;
+use Throwable;
+
+final class ReschedulingDecorator implements ReschedulingDecoratorInterface
+{
+    use DecoratorTrait;
+    use ThrowableDiagnostic\Builder\Factory\AwareTrait;
+
+    public const MAX_RESCHEDULE_DELAY_SECONDS = 86400;// (24 hours)
+
+    private /*int*/ $rescheduleDelaySeconds;
+    private /*string*/ $jobTypeCode;
+    private /*Connection*/ $connection;
+
+    public function work(): WorkerInterface
+    {
+        $this->runWorker();
+
+        if ($this->getApiV1WorkerService()->isRequestApplied()) {
+            $this->getApiV1WorkerService()
+                ->getLogger()
+                ->warning('Kojo worker state request already applied, not rescheduling.');
+        } else {
+            $this->reschedule();
+        }
+
+        return $this;
+    }
+
+    private function reschedule(): ReschedulingDecoratorInterface
+    {
+        $this->getConnection()->beginTransaction();
```

Comment:
**blocker**

Based on the other lines of code I saw (passing in the kojo `RDBMS\Connection\Service`), it looks like you intended to call `setPDO()` somewhere in this decorator, but that doesn't happen. As the code stands now, we don't know if kojo is operating within this transaction (userspace might be setting it but it's not guaranteed)

Reply:
Thank you for the feedback.
The specific kojo connection is configured in the [UserlandPdoDecorator](https://github.com/neighborhoods/KojoWorkerDecoratorComponent#userland-pdo-decorator).
The assumption is that the ReschedulingDecorator is used with the UserlandPdoDecorator and the the UserlandPdoDecorator is the outer. Would stating it explicitly in the readme be enough?

Reply:
I have made this more clear in the README.